From b947175930bb01f0112b1a556349a6c26e26fa61 Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Mon, 3 Sep 2018 17:28:40 -0400
Subject: [PATCH 1/9] Mark (v)snprintf as available

`configure` checks whether `snprintf` is available by attempting to link with
it. Windows didn't used to provide it at all. However, newer Windows does
provide it, but as an inline C implementation, so direct linking doesn't work.
For our purposes, the inline implementation is equivalent to a libc that
provides the functions, so hack our configure scripts to mark these functions
as available.
---
 configure.ac                             | 2 ++
 gettext-runtime/configure.ac             | 2 ++
 gettext-runtime/libasprintf/configure.ac | 2 ++
 gettext-tools/configure.ac               | 2 ++
 gettext-tools/examples/configure.ac      | 2 ++
 5 files changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index 78a063b..8ef1e3f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,6 +23,8 @@ AC_INIT([gettext],
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([gettext-tools/src/msgfmt.c])
 AC_CONFIG_AUX_DIR([build-aux])
+ac_cv_func_snprintf=yes
+ac_cv_func_vsnprintf=yes
 AM_INIT_AUTOMAKE([1.13 silent-rules parallel-tests dist-xz dist-lzip])
 
 dnl Override automake's tar command used for creating distributions.
diff --git a/gettext-runtime/configure.ac b/gettext-runtime/configure.ac
index 71adec3..9e3ac51 100644
--- a/gettext-runtime/configure.ac
+++ b/gettext-runtime/configure.ac
@@ -22,6 +22,8 @@ AC_INIT([gettext-runtime],
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([intl/dcigettext.c])
 AC_CONFIG_AUX_DIR([../build-aux])
+ac_cv_func_snprintf=yes
+ac_cv_func_vsnprintf=yes
 AM_INIT_AUTOMAKE([1.11.1 silent-rules parallel-tests])
 AC_CONFIG_HEADERS([config.h])
 
diff --git a/gettext-runtime/libasprintf/configure.ac b/gettext-runtime/libasprintf/configure.ac
index 736bf48..8f36a5f 100644
--- a/gettext-runtime/libasprintf/configure.ac
+++ b/gettext-runtime/libasprintf/configure.ac
@@ -20,6 +20,8 @@ AC_PREREQ([2.59])
 AC_INIT([libasprintf], [1.0])
 AC_CONFIG_SRCDIR([vasprintf.c])
 AC_CONFIG_AUX_DIR([../../build-aux])
+ac_cv_func_snprintf=yes
+ac_cv_func_vsnprintf=yes
 AM_INIT_AUTOMAKE([silent-rules])
 AC_CONFIG_HEADERS([config.h])
 
diff --git a/gettext-tools/configure.ac b/gettext-tools/configure.ac
index 3b132ed..896be75 100644
--- a/gettext-tools/configure.ac
+++ b/gettext-tools/configure.ac
@@ -22,6 +22,8 @@ AC_INIT([gettext-tools],
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([src/msgfmt.c])
 AC_CONFIG_AUX_DIR([../build-aux])
+ac_cv_func_snprintf=yes
+ac_cv_func_vsnprintf=yes
 AM_INIT_AUTOMAKE([1.11.1 silent-rules parallel-tests])
 AC_CONFIG_HEADERS([config.h])
 
diff --git a/gettext-tools/examples/configure.ac b/gettext-tools/examples/configure.ac
index c395c3d..82038a4 100644
--- a/gettext-tools/examples/configure.ac
+++ b/gettext-tools/examples/configure.ac
@@ -22,6 +22,8 @@ AC_INIT([gettext-examples],
 	[bug-gnu-gettext@gnu.org])
 AC_CONFIG_SRCDIR([installpaths.in])
 AC_CONFIG_AUX_DIR([../../build-aux])
+ac_cv_func_snprintf=yes
+ac_cv_func_vsnprintf=yes
 AM_INIT_AUTOMAKE([silent-rules])
 
 dnl Installation directories.
-- 
2.17.1

